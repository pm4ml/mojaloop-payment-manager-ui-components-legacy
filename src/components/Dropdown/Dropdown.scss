@import '../../assets/styles/vars/colors.scss';
@import '../../assets/styles/vars/shadow.scss';

.input-dropdown {
  display: inline-block;
  position: relative;
}
.input-dropdown__input {
  outline: none;
  cursor: pointer;
  box-sizing: border-box;
  border: 0;
  display: inline-block;
  flex: 0 0 auto;
  white-space: nowrap;
  user-select: none;
  text-decoration: none;
  transition: background 0.15s linear, margin 0.01s linear;
  font-style: normal;
  font-weight: 400;
  -webkit-font-smoothing: antialiased;
  border: 1px solid;
  align-items: center;
  color: $color__white;
  fill: $color__white;
  stroke: $color__white;
  border-bottom-width: 2px;

  &:active {
    color: $color__alpha-3;
    fill: $color__alpha-3;
    stroke: $color__alpha-3;
  }
}

.input-dropdown__content {
  display: flex;
  align-items: center;
  justify-content: center;
}
.input-dropdown__icon {
  display: flex;
  margin-right: 5px;
  margin-left: 0px;
  fill: inherit;
  stroke: inherit;
}

.input-dropdown__popup {
  position: absolute;
  margin-top: 5px;
  padding: 4px;
  z-index: 1;
  background-color: #f9f9f9;
  box-shadow: 0px 1px 3px rgba(0, 0, 0, 0.3);
}

.input-dropdown__mb-input--extra-small {
  padding: 2px 4px;
  font-size: 10px;
  height: 21px;
}
.input-dropdown__mb-input--small {
  padding: 6px 10px;
  font-size: 11px;
  height: 27px;
}
.input-dropdown__mb-input--medium {
  padding: 8px 13px;
  font-size: 12px;
  height: 33px;
}
.input-dropdown__mb-input--large {
  padding: 10px 16px;
  font-size: 14px;
  height: 39px;
}

.input-dropdown__mb-input--primary {
  background-color: $color__main-primary;
  border-color: $color__main-primary;

  &.noFill {
    background-color: transparent;
    color: $color__main-primary;
    fill: $color__main-primary;
    stroke: $color__main-primary;
  }

  &:active,
  &:focus {
    box-shadow: $shadow-size $shadow-color--primary;
  }

  &:hover,
  &:focus {
    background-color: $color__main-primary--active;
    border-color: $color__main-primary--active;

    &.noFill {
      color: $color__white;
      fill: $color__white;
      stroke: $color__white;
    }
  }
}
.input-dropdown__mb-input--secondary {
  background-color: $color__main-secondary;
  border-color: $color__main-secondary;

  &.noFill {
    background: transparent;
    color: $color__main-secondary;
    fill: $color__main-secondary;
    stroke: $color__main-secondary;
  }

  &:active,
  &:focus {
    box-shadow: $shadow-size $shadow-color--secondary;
  }

  &:hover,
  &:focus {
    background-color: $color__main-secondary--active;
    border-color: $color__main-secondary--active;

    &.noFill {
      color: $color__white;
      fill: $color__white;
      stroke: $color__white;
    }
  }
}

.input-dropdown__mb-input--success {
  background-color: $color__main-success;
  border-color: $color__main-success;
  color: $color__white;
  fill: $color__white;
  stroke: $color__white;

  &.noFill {
    background: transparent;
    color: $color__main-success;
    fill: $color__main-success;
    stroke: $color__main-success;
  }

  &:active,
  &:focus {
    box-shadow: $shadow-size $shadow-color--success;
  }

  &:hover,
  &:focus {
    background-color: $color__main-success--active;
    border-color: $color__main-success--active;

    &.noFill {
      color: $color__white;
      fill: $color__white;
      stroke: $color__white;
    }
  }
}
.input-dropdown__mb-input--danger {
  background-color: $color__main-danger;
  border-color: $color__main-danger;
  color: $color__white;
  fill: $color__white;
  stroke: $color__white;

  &.noFill {
    background: transparent;
    color: $color__main-danger;
    fill: $color__main-danger;
    stroke: $color__main-danger;
  }

  &:active,
  &:focus {
    box-shadow: $shadow-size $shadow-color--danger;
  }

  &:hover,
  &:focus {
    background-color: $color__main-danger--active;
    border-color: $color__main-danger--active;

    &.noFill {
      color: $color__white;
      fill: $color__white;
      stroke: $color__white;
    }
  }
}

.input-dropdown__mb-input--warning {
  background-color: $color__main-warning;
  border-color: $color__main-warning;

  &.noFill {
    background-color: transparent;
    color: $color__main-warning;
    fill: $color__main-warning;
    stroke: $color__main-warning;
  }

  &:active,
  &:focus {
    box-shadow: $shadow-size $shadow-color--warning;
  }

  &:hover,
  &:focus {
    background-color: $color__main-warning--active;
    border-color: $color__main-warning--active;

    &.noFill {
      color: $color__white;
      fill: $color__white;
      stroke: $color__white;
    }
  }
}

.input-dropdown__mb-input--dark {
  background-color: $color__alpha-5;
  border-color: transparent;
  color: $color__white;
  fill: $color__white;
  stroke: $color__white;

  &.noFill {
    background-color: transparent;
    border-color: $color__alpha-5;
    color: $color__alpha-4;
    fill: $color__alpha-4;
    stroke: $color__alpha-4;
  }

  &:active,
  &:focus {
    box-shadow: $shadow-size $shadow-color--dark;
  }

  &:hover,
  &:focus {
    background-color: $color__alpha-4;
    border-color: transparent;

    &.noFill {
      color: $color__white;
      fill: $color__white;
      stroke: $color__white;
    }
  }
}

.input-dropdown__mb-input--light {
  background-color: $color__white;
  border-color: transparent;
  color: $color__gray-5;
  fill: $color__gray-5;
  stroke: $color__gray-5;

  &.noFill {
    background-color: transparent;
    border-color: $color__white;
    color: $color__white;
    fill: $color__white;
    stroke: $color__white;
  }

  &:active,
  &:focus {
    box-shadow: $shadow-size $shadow-color--light;
  }

  &:hover,
  &:focus {
    background-color: $color__white;
    border-color: transparent;
    color: $color__gray-4;

    &.noFill {
      color: $color__gray-5;
      fill: $color__gray-5;
      stroke: $color__gray-5;
    }
  }
}

.input-dropdown__mb-input--tertiary {
  background-color: $color__main-tertiary;
  border-color: $color__main-tertiary;
  color: $color__dark-gray-2;
  fill: $color__dark-gray-2;
  stroke: $color__dark-gray-2;

  &.noFill {
    border-color: $color__gray-4;
    background: transparent;
    color: $color__gray-5;
    fill: $color__gray-5;
    stroke: $color__gray-5;
  }

  &:active,
  &:focus {
    box-shadow: $shadow-size $shadow-color--tertiary;
  }

  &:hover,
  &:focus {
    color: $color__dark-gray-2;
    fill: $color__dark-gray-2;
    stroke: $color__dark-gray-2;
    background-color: $color__main-tertiary--active;
    border-color: $color__main-tertiary--active;
  }
}

.input-dropdown__mb-input--disabled {
  background-color: $color__gray-2;
  border-color: $color__gray-2;
  color: $color__dark-gray-1;
  fill: $color__gray-5;
  stroke: $color__gray-5;

  &.noFill {
    border-color: $color__gray-2;
    background: transparent;
    color: $color__gray-5;
    fill: $color__gray-5;
    stroke: $color__gray-5;
  }
}

.input-dropdown__mb-input--primary,
.input-dropdown__mb-input--secondary,
.input-dropdown__mb-input--tertiary,
.input-dropdown__mb-input--danger,
.input-dropdown__mb-input--warning,
.input-dropdown__mb-input--dark,
.input-dropdown__mb-input--light {
  &:hover {
    &.noFill {
      &:active {
        color: $color__alpha-3;
        fill: $color__alpha-3;
        stroke: $color__alpha-3;
      }
    }
  }
  &:active {
    color: $color__alpha-3;
    fill: $color__alpha-3;
    stroke: $color__alpha-3;
  }
}
