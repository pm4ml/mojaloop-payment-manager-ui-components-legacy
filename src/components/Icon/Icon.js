import './Icon.scss';

import PropTypes from 'prop-types';
import React, { PureComponent } from 'react';

import * as utils from '../../utils/common';
import Tooltip from '../Tooltip';

const iconSizes = {
  xs: 10,
  s: 12,
  m: 14,
  l: 16,
};

class Icon extends PureComponent {
  constructor(props) {
    super(props);
    this.onClick = this.onClick.bind(this);
  }

  onClick(e) {
    if (this.props.onClick) {
      this.props.onClick(e);
    }
  }
  render() {
    const {
      id,
      className,
      style,
      size,
      name,
      fill,
      stroke,
      spin,
      tooltip,
      tooltipKind,
      tooltipDelay,
      tooltipPosition,
    } = this.props;

    const svgStyle = {
      height: `${size}px`,
      width: `${size}px`,
      fill,
      stroke,
      ...style,
    };
    const componentClassName = utils.composeClassNames([
      'el-icon',
      spin && 'el-icon--spin',
      className,
    ]);
    const svg = (
      <svg id={id} style={svgStyle} onClick={this.onClick} className={componentClassName}>
        <use xlinkHref={`#${name}`} />
      </svg>
    );
    if (tooltip) {
      return (
        <Tooltip label={tooltip} position={tooltipPosition} delay={tooltipDelay} kind={tooltipKind}>
          {svg}
        </Tooltip>
      );
    }
    return svg;
  }
}

Icon.propTypes = {
  id: PropTypes.string,
  style: PropTypes.shape(),
  className: PropTypes.string,
  size: PropTypes.number,
  name: PropTypes.string,
  fill: PropTypes.string,
  stroke: PropTypes.string,
  spin: PropTypes.bool,
  onClick: PropTypes.func,
  tooltipKind: PropTypes.oneOf(['regular', 'error', 'info', 'warning', 'neutral']),
  tooltipDelay: PropTypes.number,
  tooltip: PropTypes.oneOfType([PropTypes.node, PropTypes.string]),
  tooltipPosition: PropTypes.oneOf(['top', 'bottom', 'left', 'right']),
};
Icon.defaultProps = {
  id: undefined,
  style: undefined,
  className: undefined,
  size: 20,
  name: undefined,
  fill: undefined,
  stroke: undefined,
  spin: false,
  onClick: undefined,
  tooltipKind: undefined,
  tooltipDelay: undefined,
  tooltip: undefined,
  tooltipPosition: undefined,
};

export default Icon;
export { iconSizes };
