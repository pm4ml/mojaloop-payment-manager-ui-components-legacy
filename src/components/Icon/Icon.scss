@import '../../assets/styles/vars/colors.scss';

.el-icon {
  display: flex;
  flex: 0 0 auto;
}

.el-icon--spin {
  animation: spinIcon 1s linear infinite;
}

@keyframes spinIcon {
  0% {
    transform: rotate(0deg);
  }

  100% {
    transform: rotate(360deg);
  }
}
