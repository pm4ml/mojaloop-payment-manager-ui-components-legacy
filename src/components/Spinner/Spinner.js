import './Spinner.scss';

import PropTypes from 'prop-types';
import React, { PureComponent } from 'react';

const polarToCartesian = (centerX, centerY, radius, angleInDegrees) => {
  const angleInRadians = (angleInDegrees - 90) * (Math.PI / 180.0);
  return {
    x: centerX + radius * Math.cos(angleInRadians),
    y: centerY + radius * Math.sin(angleInRadians),
  };
};

const describeArc = (x, y, radius, startAngle, endAngle) => {
  const start = polarToCartesian(x, y, radius, endAngle);
  const end = polarToCartesian(x, y, radius, startAngle);
  const largeArcFlag = endAngle - startAngle <= 180 ? '0' : '1';
  return ['M', start.x, start.y, 'A', radius, radius, 0, largeArcFlag, 0, end.x, end.y].join(' ');
};

class Spinner extends PureComponent {
  render() {
    const { size, center, color } = this.props;
    const realSize = typeof size === 'string' ? { s: 20, m: 40, l: 60 }[size] : size;
    const strokeWidth = realSize / 10;
    const width = `${realSize}px`;
    const height = `${realSize}px`;
    const position = realSize / 2;
    const radius = position - strokeWidth;
    const spinnerStyle = {
      width: center === false && width,
      height: center === false && height,
    };
    const pathStyle = { stroke: color };

    return (
      <div className={`el-spinner ${center ? 'center' : ''}`} style={spinnerStyle}>
        <svg
          className="el-spinner__component"
          width={width}
          height={height}
          viewBox={`0 0 ${realSize} ${realSize}`}
        >
          <path
            className="el-spinner__svg-path"
            strokeWidth={strokeWidth}
            d={describeArc(position, position, radius, 90, 200)}
            style={pathStyle}
          />
        </svg>
      </div>
    );
  }
}

Spinner.defaultProps = {
  size: 20,
  center: false,
  color: undefined,
};

Spinner.propTypes = {
  size: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
  center: PropTypes.bool,
  color: PropTypes.string,
};
export default Spinner;
