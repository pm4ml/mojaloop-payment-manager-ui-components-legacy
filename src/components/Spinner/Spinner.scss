@import '../../assets/styles/vars/colors.scss';

.el-spinner {
  animation: rotateSpinner 4s linear infinite;

  &.center {
    display: flex;
    width: 100%;
    height: inherit;
    align-items: center;
    justify-content: center;
  }
}
.el-spinner__component {
  animation: rotateSpinner 0.8s cubic-bezier(0.53, 0.26, 0.54, 0.74) infinite;
}

.el-spinner__svg-path {
  fill: none;
  stroke-dasharray: 100;
  stroke-dashoffset: 0;
  transform-origin: center;
  stroke-linecap: round;
  stroke: $color__main-primary;
}

@keyframes rotateSpinner {
  0% {
    transform: rotate(0deg);
  }
  100% {
    transform: rotate(360deg);
  }
}
