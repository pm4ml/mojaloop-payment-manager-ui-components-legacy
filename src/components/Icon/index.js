import Icon, { iconSizes } from './Icon';

export default Icon;
export { iconSizes };
