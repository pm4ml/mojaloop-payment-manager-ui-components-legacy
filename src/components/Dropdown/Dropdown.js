import './Dropdown.scss';

import PropTypes from 'prop-types';
import React, { PureComponent } from 'react';

import * as utils from '../../utils/common';
import Icon, { iconSizes } from '../Icon';
import Spinner from '../Spinner';
import Tooltip from '../Tooltip';

class Dropdown extends PureComponent {
  constructor(props) {
    super(props);
    this.onClick = this.onClick.bind(this);
    this.testKey = this.testKey.bind(this);
    this.handlePageClick = this.handlePageClick.bind(this);
    this.state = {
      isOpen: false,
    };
  }

  componentDidMount() {
    window.addEventListener('mouseup', this.handlePageClick, false);
  }
  componentWillUnmount() {
    window.removeEventListener('mouseup', this.handlePageClick, false);
  }

  onClick() {
    if (this.props.disabled) {
      return;
    }

    this.setState({
      isOpen: !this.state.isOpen,
    });
  }
  handlePageClick(e) {
    if (!this.container.contains(e.target)) {
      this.setState({ isOpen: false });
    }
  }
  testKey(e) {
    if (e.nativeEvent.keyCode === 9) {
      e.preventDefault();
      utils.focusNextFocusableElement(this.input, !e.nativeEvent.shiftKey);
      return;
    }

    if (e.nativeEvent.keyCode === 13) {
      e.preventDefault();
      this.onClick(e);
    }
  }
  render() {
    const {
      id,
      className,
      style,
      size,
      kind,
      label,
      icon,
      noFill,
      disabled,
      pending,
      tooltip,
    } = this.props;
    const isDisabledOrPending = disabled === true || pending === true;
    const iconSize = iconSizes[size];
    const classNames = utils.composeClassNames([
      className,
      'mb-input',
      'input-dropdown__input',
      size === 's' && 'input-dropdown__mb-input--small',
      size === 'm' && 'input-dropdown__mb-input--medium',
      size === 'l' && 'input-dropdown__mb-input--large',
      kind === 'primary' && 'input-dropdown__mb-input--primary',
      kind === 'secondary' && 'input-dropdown__mb-input--secondary',
      kind === 'tertiary' && 'input-dropdown__mb-input--tertiary',
      kind === 'success' && 'input-dropdown__mb-input--success',
      kind === 'danger' && 'input-dropdown__mb-input--danger',
      kind === 'warning' && 'input-dropdown__mb-input--warning',
      kind === 'dark' && 'input-dropdown__mb-input--dark',
      kind === 'light' && 'input-dropdown__mb-input--light',
      isDisabledOrPending && 'mb-input--disabled input-dropdown__mb-input--disabled',
      pending && 'mb-input--pending input-dropdown__mb-input--pending',
      noFill && 'noFill',
    ]);

    const button = (
      <div
        className="input-dropdown"
        ref={container => {
          this.container = container;
        }}
      >
        <button
          ref={input => {
            this.input = input;
          }}
          id={id}
          style={style}
          className={classNames}
          onKeyDown={this.testKey}
          onClick={this.onClick}
          disabled={isDisabledOrPending}
          label={label}
          kind={kind}
        >
          <div className="input-dropdown__content">
            {(pending || icon) && (
              <div className="input-dropdown__icon">
                {pending ? (
                  <Spinner color="inherit" size={iconSize} />
                ) : (
                  <Icon name={icon} stroke="none" spin={pending} size={iconSize} />
                )}
              </div>
            )}
            {label && <span>{label}</span>}
          </div>
        </button>
        {this.state.isOpen && <div className="input-dropdown__popup">{this.props.children}</div>}
      </div>
    );

    if (tooltip) {
      return (
        <Tooltip label={tooltip} position="top">
          {button}
        </Tooltip>
      );
    }
    return button;
  }
}
Dropdown.propTypes = {
  className: PropTypes.string,
  id: PropTypes.string,
  style: PropTypes.shape(),
  kind: PropTypes.oneOf([
    'primary',
    'secondary',
    'tertiary',
    'success',
    'danger',
    'warning',
    'dark',
    'light',
  ]),
  size: PropTypes.oneOf(['s', 'm', 'l']),
  label: PropTypes.string,
  icon: PropTypes.string,
  noFill: PropTypes.bool,
  disabled: PropTypes.bool,
  pending: PropTypes.bool,
  tooltip: PropTypes.string,
};
Dropdown.defaultProps = {
  className: undefined,
  id: undefined,
  style: undefined,
  kind: 'primary',
  size: 'l',
  label: undefined,
  icon: undefined,
  noFill: false,
  disabled: false,
  pending: false,
  tooltip: undefined,
};

export default Dropdown;
