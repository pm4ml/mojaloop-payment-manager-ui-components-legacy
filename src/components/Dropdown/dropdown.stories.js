/* eslint no-console: "off" */
import React from 'react';
import Row from '../Row';
import Column from '../Column';
import Dropdown from './Dropdown';

export default {
  title: 'Dropdown',
  component: Dropdown,
};

const Content = () => (
  <Column style={{ width: '200px' }}>
    <Row>This is the header</Row>
    <Row>This is the content</Row>
  </Column>
);

export const AllKinds = () => (
  <Row align="space-between center">
    <Dropdown label="Primary" kind="primary">
      <Content />
    </Dropdown>
    <Dropdown label="Secondary" kind="secondary">
      <Content />
    </Dropdown>
    <Dropdown label="Tertiary" kind="tertiary">
      <Content />
    </Dropdown>
    <Dropdown label="Success" kind="success">
      <Content />
    </Dropdown>
    <Dropdown label="Danger" kind="danger">
      <Content />
    </Dropdown>
    <Dropdown label="Warning" kind="warning">
      <Content />
    </Dropdown>
    <Dropdown label="Dark" kind="dark">
      <Content />
    </Dropdown>
    <Dropdown label="Light" kind="light">
      <Content />
    </Dropdown>
  </Row>
);

export const AllKindsWithNoFill = () => (
  <Row align="space-between center" className="p10 bg-dark">
    <Dropdown noFill label="Primary" kind="primary">
      <Content />
    </Dropdown>
    <Dropdown noFill label="Secondary" kind="secondary">
      <Content />
    </Dropdown>
    <Dropdown noFill label="Tertiary" kind="tertiary">
      <Content />
    </Dropdown>
    <Dropdown noFill label="Success" kind="success">
      <Content />
    </Dropdown>
    <Dropdown noFill label="Danger" kind="danger">
      <Content />
    </Dropdown>
    <Dropdown noFill label="Warning" kind="warning">
      <Content />
    </Dropdown>
    <Dropdown noFill label="Dark" kind="dark">
      <Content />
    </Dropdown>
    <Dropdown noFill label="Light" kind="light">
      <Content />
    </Dropdown>
  </Row>
);

export const AllKindsWithIcon = () => (
  <Row align="space-between center">
    <Dropdown icon="deploy-small" label="Primary" kind="primary">
      <Content />
    </Dropdown>
    <Dropdown icon="deploy-small" label="Secondary" kind="secondary">
      <Content />
    </Dropdown>
    <Dropdown icon="deploy-small" label="Tertiary" kind="tertiary">
      <Content />
    </Dropdown>
    <Dropdown icon="deploy-small" label="Success" kind="success">
      <Content />
    </Dropdown>
    <Dropdown icon="deploy-small" label="Danger" kind="danger">
      <Content />
    </Dropdown>
    <Dropdown icon="deploy-small" label="Warning" kind="warning">
      <Content />
    </Dropdown>
    <Dropdown icon="deploy-small" label="Dark" kind="dark">
      <Content />
    </Dropdown>
    <Dropdown icon="deploy-small" label="Light" kind="light">
      <Content />
    </Dropdown>
  </Row>
);

export const AllKindsWithNoFillWithIcon = () => (
  <Row align="space-between center" className="p10 bg-dark">
    <Dropdown noFill icon="deploy-small" label="Primary" kind="primary">
      <Content />
    </Dropdown>
    <Dropdown noFill icon="deploy-small" label="Secondary" kind="secondary">
      <Content />
    </Dropdown>
    <Dropdown noFill icon="deploy-small" label="Tertiary" kind="tertiary">
      <Content />
    </Dropdown>
    <Dropdown noFill icon="deploy-small" label="Success" kind="success">
      <Content />
    </Dropdown>
    <Dropdown noFill icon="deploy-small" label="Danger" kind="danger">
      <Content />
    </Dropdown>
    <Dropdown noFill icon="deploy-small" label="Warning" kind="warning">
      <Content />
    </Dropdown>
    <Dropdown noFill icon="deploy-small" label="Dark" kind="dark">
      <Content />
    </Dropdown>
    <Dropdown noFill icon="deploy-small" label="Light" kind="light">
      <Content />
    </Dropdown>
  </Row>
);

export const AllPending = () => (
  <Row align="space-between center">
    <Dropdown label="Primary" kind="primary" pending>
      <Content />
    </Dropdown>
    <Dropdown label="Secondary" kind="secondary" pending>
      <Content />
    </Dropdown>
    <Dropdown label="Tertiary" kind="tertiary" pending>
      <Content />
    </Dropdown>
    <Dropdown label="Success" kind="success" pending>
      <Content />
    </Dropdown>
    <Dropdown label="Danger" kind="danger" pending>
      <Content />
    </Dropdown>
    <Dropdown label="Warning" kind="warning" pending>
      <Content />
    </Dropdown>
    <Dropdown label="Dark" kind="dark" pending>
      <Content />
    </Dropdown>
    <Dropdown label="Light" kind="light" pending>
      <Content />
    </Dropdown>
    <Dropdown label="Disabled" disabled pending>
      <Content />
    </Dropdown>
    <Dropdown label="Icon" icon="deploy-small" pending>
      <Content />
    </Dropdown>
  </Row>
);

export const AllPendingWithNoFill = () => (
  <Row align="space-between center">
    <Dropdown noFill label="Primary" kind="primary" pending>
      <Content />
    </Dropdown>
    <Dropdown noFill label="Secondary" kind="secondary" pending>
      <Content />
    </Dropdown>
    <Dropdown noFill label="Tertiary" kind="tertiary" pending>
      <Content />
    </Dropdown>
    <Dropdown noFill label="Success" kind="success" pending>
      <Content />
    </Dropdown>
    <Dropdown noFill label="Danger" kind="danger" pending>
      <Content />
    </Dropdown>
    <Dropdown noFill label="Warning" kind="warning" pending>
      <Content />
    </Dropdown>
    <Dropdown noFill label="Dark" kind="dark" pending>
      <Content />
    </Dropdown>
    <Dropdown noFill label="Light" kind="light" pending>
      <Content />
    </Dropdown>
    <Dropdown noFill label="Disabled" disabled pending>
      <Content />
    </Dropdown>
    <Dropdown noFill label="Icon" icon="deploy-small" pending>
      <Content />
    </Dropdown>
  </Row>
);

export const AllDisabled = () => (
  <Row align="space-between center">
    <Dropdown label="Primary" kind="primary" disabled>
      <Content />
    </Dropdown>
    <Dropdown label="Secondary" kind="secondary" disabled>
      <Content />
    </Dropdown>
    <Dropdown label="Tertiary" kind="tertiary" disabled>
      <Content />
    </Dropdown>
    <Dropdown label="Success" kind="success" disabled>
      <Content />
    </Dropdown>
    <Dropdown label="Danger" kind="danger" disabled>
      <Content />
    </Dropdown>
    <Dropdown label="Warning" kind="warning" disabled>
      <Content />
    </Dropdown>
    <Dropdown label="Dark" kind="dark" disabled>
      <Content />
    </Dropdown>
    <Dropdown label="Pending" pending disabled>
      <Content />
    </Dropdown>
    <Dropdown label="Icon" icon="deploy-small" disabled>
      <Content />
    </Dropdown>
    <Dropdown label="No Fill" icon="deploy-small" noFill disabled>
      <Content />
    </Dropdown>
    <Dropdown label="No Fill" icon="deploy-small" noFill kind="secondary" disabled>
      <Content />
    </Dropdown>
    <Dropdown label="No Fill" icon="deploy-small" noFill kind="tertiary" disabled>
      <Content />
    </Dropdown>
    <Dropdown label="No Fill" icon="deploy-small" noFill kind="danger" disabled>
      <Content />
    </Dropdown>
  </Row>
);

export const AllSizes = () => (
  <div>
    <Row align="space-between center">
      <Dropdown label="Large size " size="l">
        <Content />
      </Dropdown>
      <Dropdown label="Medium size " size="m">
        <Content />
      </Dropdown>
      <Dropdown label="Small size " size="s">
        <Content />
      </Dropdown>
      <Dropdown label="Large size and icon" icon="deploy-small" size="l">
        <Content />
      </Dropdown>
      <Dropdown label="Medium size and icon" icon="deploy-small" size="m">
        <Content />
      </Dropdown>
      <Dropdown label="Small size and icon" icon="deploy-small" size="s">
        <Content />
      </Dropdown>
    </Row>
    <Row align="space-between center">
      <Dropdown label="Large size pending " pending size="l">
        <Content />
      </Dropdown>
      <Dropdown label="Medium size pending " pending size="m">
        <Content />
      </Dropdown>
      <Dropdown label="Small size pending " pending size="s">
        <Content />
      </Dropdown>
    </Row>
  </div>
);

export const WithTooltip = () => (
  <Row align="space-between center">
    <Dropdown label="Test Dropdown Tooltip" tooltip="Test!">
      <Content />
    </Dropdown>
  </Row>
);
