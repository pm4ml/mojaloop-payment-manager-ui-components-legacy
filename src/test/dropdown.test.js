import { mount, shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import React from 'react';

import Dropdown from '../components/Dropdown';
import Icon from '../components/Icon';
import Spinner from '../components/Spinner';

// Shallow render, no sub-components ( still available children html)

it('renders the label', () => {
  const wrapper = shallow(<Dropdown label="Test-Dropdown" />);
  expect(wrapper.find('span').text()).toBe('Test-Dropdown');
});

it('renders the disabled state', () => {
  const wrapper = shallow(<Dropdown label="Click" disabled />);
  expect(wrapper.find('button').prop('disabled')).toBe(true);
});

it('renders the spinner and sets disabled', () => {
  const wrapper = shallow(<Dropdown pending />);
  expect(wrapper.find('.input-dropdown__icon')).toHaveLength(1);
  expect(wrapper.find(Spinner)).toHaveLength(1);
  expect(wrapper.find('button').prop('disabled')).toBe(true);
});

it('renders spinner with precedence to icon', () => {
  const wrapper = shallow(<Dropdown icon="deploy" pending />);
  expect(wrapper.find(Spinner)).toHaveLength(1);
  expect(wrapper.find(Icon)).toHaveLength(0);
});

it('renders the dropdown icon', () => {
  const wrapper = shallow(<Dropdown icon="deploy" />);
  expect(wrapper.find('.input-dropdown__icon')).toHaveLength(1);
  expect(wrapper.find(Icon)).toHaveLength(1);
});

it('applies the prop className', () => {
  const wrapper = shallow(<Dropdown className="test" />);
  expect(wrapper.find('.test')).toHaveLength(1);
});

it('renders the prop id', () => {
  const wrapper = shallow(<Dropdown id="testDropdownId" />);
  expect(wrapper.find('#testDropdownId')).toHaveLength(1);
});

it('renders the id on the "button" element', () => {
  const wrapper = shallow(<Dropdown id="testDropdownId" />);
  expect(wrapper.find('button').prop('id')).toBe('testDropdownId');
});

it('renders the default correct kind', () => {
  const wrapper = shallow(<Dropdown />);
  expect(wrapper.find('button').prop('kind')).toBe('primary');
});

it('renders all the kinds', () => {
  ['primary', 'secondary', 'tertiary', 'danger', 'warning', 'dark'].forEach(kind => {
    const wrapper = shallow(<Dropdown kind={kind} />);
    expect(wrapper.find('button').prop('kind')).toBe(kind);
  });
});

it('renders all the "noFill" prop', () => {
  const wrapper = shallow(<Dropdown noFill />);
  expect(wrapper.find('.noFill')).toHaveLength(1);
});

it('renders the large, medium, small sizes', () => {
  const sizes = {
    large: 'l',
    medium: 'm',
    small: 's',
  };
  Object.entries(sizes).forEach(([name, size]) => {
    const wrapper = shallow(<Dropdown size={size} />);
    const className = `input-dropdown__mb-input--${name}`;
    expect(wrapper.find('.mb-input').hasClass(className)).toBeTruthy();
  });
});

it('renders multiple props', () => {
  const wrapper = shallow(
    <Dropdown label="Test" icon="x" kind="primary" disabled pending noFill />,
  );
  expect(wrapper.find('span').text()).toBe('Test');
  expect(wrapper.find('button').prop('kind')).toBe('primary');
  expect(wrapper.find('button').prop('disabled')).toBe(true);
  expect(wrapper.find(Spinner)).toHaveLength(1);
  expect(wrapper.find(Icon)).toHaveLength(0);
});

// Mount render, include sub-components

it('renders the inner icon component', () => {
  const wrapper = mount(<Dropdown icon="deploy" />);
  expect(wrapper.find('.el-icon')).toHaveLength(1);
});

it('renders the inner spinner component', () => {
  const wrapper = mount(<Dropdown pending />);
  expect(wrapper.find('.el-spinner')).toHaveLength(1);
});

it('renders the spinner component and overrides the icon prop', () => {
  const wrapper = mount(<Dropdown pending icon="deploy" />);
  expect(wrapper.find('.el-icon')).toHaveLength(0);
  expect(wrapper.find('.el-spinner')).toHaveLength(1);
});

it('does not render the popup when not clicked', () => {
  const wrapper = mount(
    <Dropdown>
      <span>Content!</span>
    </Dropdown>,
  );
  expect(wrapper.find('.input-dropdown__popup').exists()).toBe(false);
});

it('renders the popup when clicked', () => {
  const wrapper = mount(
    <Dropdown>
      <span>Content!</span>
    </Dropdown>,
  );
  wrapper.find('button').simulate('click');
  expect(wrapper.find('.input-dropdown__popup').exists()).toBe(true);
});

it('closes the popup when clicking again the button', () => {
  const wrapper = mount(
    <Dropdown>
      <span>Content!</span>
    </Dropdown>,
  );
  wrapper.find('button').simulate('click');
  expect(wrapper.find('.input-dropdown__popup').exists()).toBe(true);
  wrapper.find('button').simulate('click');
  expect(wrapper.find('.input-dropdown__popup').exists()).toBe(false);
});

// Snapshot testing

it('renders the dropdown correctly when multiple props are set', () => {
  const wrapper = shallow(<Dropdown label="Snapshot dropdown" kind="secondary" disabled pending />);
  expect(shallowToJson(wrapper)).toMatchSnapshot();
});
