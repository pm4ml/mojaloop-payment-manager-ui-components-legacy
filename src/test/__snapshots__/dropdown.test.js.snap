// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the dropdown correctly when multiple props are set 1`] = `
<div
  className="input-dropdown"
>
  <button
    className="mb-input input-dropdown__input input-dropdown__mb-input--large input-dropdown__mb-input--secondary mb-input--disabled input-dropdown__mb-input--disabled mb-input--pending input-dropdown__mb-input--pending"
    disabled={true}
    kind="secondary"
    label="Snapshot dropdown"
    onClick={[Function]}
    onKeyDown={[Function]}
  >
    <div
      className="input-dropdown__content"
    >
      <div
        className="input-dropdown__icon"
      >
        <Spinner
          center={false}
          color="inherit"
          size={16}
        />
      </div>
      <span>
        Snapshot dropdown
      </span>
    </div>
  </button>
</div>
`;
